tiledb/cf/netcdf_engine/_array_converters.py

```diff
@@ -124,62 +187,45 @@ def copy(
                     dim_creator.base.copy_metadata(netcdf_group, tiledb_array)
-            # Copy array data to TileDB.
-            if self.sparse:
-                shape: Optional[Union[int, Sequence[int]]] = -1
-            else:
-                shape = (
-                    None
-                    if all(
-                        isinstance(dim_creator.base, NetCDF4ToDimConverter)
-                        for dim_creator in self._domain_creator
-                    )
-                    else self.domain_creator.get_dense_query_shape(netcdf_group)
-                )
-            data = {}
-            for attr_creator in self:
-                if isinstance(attr_creator, NetCDF4ToAttrConverter):
-                    data[attr_creator.name] = attr_creator.get_values(
-                        netcdf_group, sparse=self.sparse, shape=shape
-                    )
-                else:
-                    if (
-                        assigned_attr_values is None
-                        or attr_creator.name not in assigned_attr_values
-                    ):
-                        raise KeyError(
-                            f"Missing value for attribute '{attr_creator.name}'."
-                        )
-                    data[attr_creator.name] = assigned_attr_values[attr_creator.name]
-            coord_values = self._domain_creator.get_query_coordinates(
-                netcdf_group, self.sparse, assigned_dim_values
+            # Copy array data for first fragment.
+            self._copy_to_array(
+                netcdf_group,
+                tiledb_array,
+                next(fragment_indexers),
+                assigned_dim_values,
+                assigned_attr_values,
             )
-            tiledb_array[coord_values] = data
+        # Copy array data for remaining fragments.
+        for indexer in fragment_indexers:
+            with tiledb.open(
+                tiledb_uri, mode="w", key=tiledb_key, ctx=tiledb_ctx
+            ) as tiledb_array:
+                self._copy_to_array(
+                    netcdf_group,
+                    tiledb_array,
+                    indexer,
+                    assigned_dim_values,
+                    assigned_attr_values,
+                )
 
 
 class NetCDF4DomainConverter(DomainCreator):
     """Converter for NetCDF dimensions to a TileDB domain."""
 
-    def get_dense_query_shape(self, netcdf_group: netCDF4.Dataset) -> Tuple[int, ...]:
-        """Returns the shape of the coordinates for copying from the requested NetCDF
-        group to a dense array.
-
-        Parameters:
-            netcdf_group: Group to query the data from.
-        """
-        return tuple(
-            dim_creator.base.get_query_size(netcdf_group)
-            if isinstance(dim_creator.base, NetCDF4ToDimConverter)
-            else 1
-            for dim_creator in self
+    def get_fragment_indexers(self, netcdf_group: netCDF4.Dataset):
```

Comment:
- Missing return type annotation
- For iterators/generators `iter_` is commonly preferred over `get_`
- Internal method, better prefixed with `_` (or just inline it, it's called only once and it's one line).